src/components/dashboard/FeedItems/EventCounterParty.js

```diff
@@ -1,32 +1,37 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { get } from 'lodash'
 import { Text } from '../../common'
 import { withStyles } from '../../../lib/styles'
+import useProfile from '../../../lib/userStorage/useProfile'
 
-const EventCounterParty = ({ feedItem, styles, style, textStyle, subtitle, isSmallDevice }) => {
-  let direction = ''
-  let itemSubtitle = get(feedItem, 'data.subtitle', '')
-  let displayText = itemSubtitle && subtitle ? itemSubtitle : get(feedItem, 'data.endpoint.displayName')
+const useEventDirection = (feedItemType, reverse = false) => {
```

Comment:
1. not sure we need it as effect. it could be just the utility function. you're free to wrap its call to useMemo in any component/hook if you need this
2. we have some utils in `src/lib/userStorage/FeedStorage.js`
3. also there're contstants defined for claim (in the file linked above), receive etc. need to re-use them